nixos/modules/config/sysctl.nix

```diff
@@ -34,14 +42,373 @@ in
+            default = null;
+          };
+
+          "dev.tty.ldisc_autoload" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Automatically load line discipline modules for tty devices.";
+            # Prevent unprivileged attackers from loading vulnerable line disciplines with the TIOCSETD ioctl
+            default = false;
+          };
+
+          "fs.protected_hardlinks" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Protect hardlinks created by other users.";
+            # Prevent unprivileged attackers from creating world-writable sticky hardlinks
+            default = true;
+          };
+
+          "fs.protected_symlinks" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Protect symlinks created by other users.";
+            # Prevent unprivileged attackers from creating world-writable sticky symlinks
+            default = true;
+          };
+
+          "fs.protected_fifos" = mkOption {
+            type = types.nullOr highestValueType;
+            description = "Protect pipes and FIFOs from being written to by other users.";
+            # Prevent unprivileged attackers from creating world-writable sticky FIFOs
+            default = 2;
+          };
+
+          "fs.protected_regular" = mkOption {
+            type = types.nullOr highestValueType;
+            description = "Protect files created by other users.";
+            # Prevent unprivileged attackers from creating world-writable sticky files
+            default = 2;
+          };
+
+          "fs.suid_dumpable" = mkOption {
+            type = types.nullOr highestValueType;
+            description = "Controls the core dump mode for setuid processes.";
+            # Prevent setuid processes from dumping core
+            default = 0;
+          };
+
+          "kernel.ctrl-alt-del" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Enable the Ctrl-Alt-Del key combination to trigger a kernel reboot.";
+            # This is a security risk, as it allows users to reboot the system without authentication
+            default = false;
+          };
+
+          "kernel.dmesg_restrict" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Restrict access to kernel log buffer.";
+            # Hide kernel log buffer to make it harder to exploit kernel vulnerabilities (e.g. by leaking kernel pointers)
+            default = true;
+          };
+
+          "kernel.kptr_restrict" = mkOption {
+            type = types.nullOr highestValueType;
+            description = "Restrict access to kernel pointers.";
+            # Hide kernel pointers to make it harder to exploit kernel vulnerabilities
+            default = 2;
+          };
+
+          "kernel.modules_disabled" = mkOption {
```

Comment:
How different is this from [`security.lockKernelModules`](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/security/lock-kernel-modules.nix)?